### PR TITLE
all: Modular Messaging rename

### PR DESCRIPTION
**Pull request by evan-gray**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,18 +19,18 @@ See [DEVELOP.md](./DEVELOP.md).
...
 
-- A non-upgradeable, non-administrated/owned **Router** which manages integrator Transceiver configuration and message state. This abstracts Transceivers from Integrators. The goal of the Router is to provide strictly the functionality necessary for agnostically generating and aggregating attestations across multiple protocols.
-- An opinionated standard for **Transceiver** design. The intention is for these to be safe to share and trust across integrators. The goal of a Transceiver is to interface between the Router and a particular messaging / attestation protocol.
+- A non-upgradeable, non-administrated/owned **Endpoint** which manages integrator Adapter configuration and message state. This abstracts Adapters from Integrators. The goal of the Endpoint is to provide strictly the functionality necessary for agnostically generating and aggregating attestations across multiple protocols.
+- An opinionated standard for **Adapter** design. The intention is for these to be safe to share and trust across integrators. The goal of a Adapter is to interface between the Endpoint and a particular messaging / attestation protocol.
```
Comment (bruce-riley) [inline README.md:33]: Nit: "an Adapter".

File: README.md [hunk 2]
```diff
@@ -45,11 +45,11 @@ Any implementation should prioritize
...
 
-- Handling complex thresholding logic in the Router.
+- Handling complex thresholding logic in the Endpoint.
   - No one size fits all solution. Instead, expose a way for Integrators to define their own rules on their side.
-- Managing peers in the Router.
-  - This would introduce Integrator/application-specific logic into the Router and make it more difficult for an Integrator to migrate to a new Router. It is more appropriate for accepted peers to be configured at the application level (though standard libraries can be provided for this).
-- Pass-through parameters specific to a Transceiver.
+- Managing peers in the Endpoint.
+  - This would introduce Integrator/application-specific logic into the Endpoint and make it more difficult for an Integrator to migrate to a new Endpoint. It is more appropriate for accepted peers to be configured at the application level (though standard libraries can be provided for this).
+- Pass-through parameters specific to a Adapter.
```
Comment (bruce-riley) [inline README.md:52]: Maybe globally search for "a adapter"?

File: README.md [hunk 3]
```diff
@@ -320,88 +320,88 @@ The Router MUST contain the following functionality for an Admin
...
... used by an integrator. Today, only the [WormholeAdapter](https://github.com/wormhole-foundation/example-native-token-transfers/blob/b2ecf88c8840de7335b035b24d7f7a299862e2a6/evm/src/Adapter/WormholeAdapter/WormholeAdapter.sol#L128-L140) supports a custom AdapterInstruction, which is used to skip relaying (if it is configured and the corresponding byte is set). Since this design explicitly states that Adapters SHOULD avoid branching logic and the Endpoint should offer a uniform experience, this field has been dropped. This may impact future Adapters which desire to use additional information which is not provided by the interface - examples may include an Axelar Adapter which uses off-chain gas quotes or Wormhole Standard Relayer Adapter which uses signed off-chain gas quotes. It should be noted again that, unlike the existing NTT design, these Adapters are ONLY for relaying attestations. It is, however, possible for those use cases to be accommodated by the Integrator explicitly calling a method on the Adapter directly, interim state being stored, and read/cleared when `sendMessage` is called, though this once again bleeds Adapter implementation detail into the Integrator contract.
```
Comment (bruce-riley) [inline README.md:372]: Since the existing NTT uses "Transceiver", should that stay unchanged here?

File: svm/programs/endpoint/src/lib.rs [hunk 4]
```diff
@@ -26,80 +26,68 @@ pub mod router {
...
-        ctx: Context<EnableTransceiver>,
-        args: TransceiverInfoArgs,
-    ) -> Result<()> {
-        instructions::enable_transceiver::enable_recv_transceiver(ctx, args)
+    pub fn enable_recv_adapter(ctx: Context<EnableAdapter>, args: AdapterInfoArgs) -> Result<()> {
+        instructions::enable_adapter::enable_recv_adapter(ctx, args)
     }
 
-    /// Sets a transceiver as a send transceiver for a specific chain
+    /// Sets a adapter as a send adapter for a specific chain
```
Comment (bingyuyap) [inline svm/programs/endpoint/src/lib.rs:54]:
```suggestion
    /// Sets an adapter as a send adapter for a specific chain
```

File: svm/programs/mock-adapter/README.md [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
 
 ## Overview
 
-The mock transceiver program is crucial for testing the router program functionality. It serves as a gateway to test several features of the router program because:
+The mock adapter program is crucial for testing the endpoint program functionality. It serves as a gateway to test several features of the endpoint program because:
 
 1. It simulates the message pickup process, which is essential for sending cross-chain messages.
 2. It demonstrates how to attest to messages.
-3. It allows us to test the router program's handling of Cross-Program Invocations (CPIs) from a transceiver.
+3. It allows us to test the endpoint program's handling of Cross-Program Invocations (CPIs) from a adapter.
```
Comment (bingyuyap) [inline svm/programs/mock-adapter/README.md:11]:
```suggestion
3. It allows us to test the endpoint program's handling of Cross-Program Invocations (CPIs) from an adapter.
```

File: svm/programs/mock-adapter/src/lib.rs [hunk 6]
```diff
@@ -1,58 +1,58 @@
...
+pub mod mock_adapter {
 
     use super::*;
 
-    /// Invokes the pick_up_message instruction on the router program via CPI
+    /// Invokes the pick_up_message instruction on the endpoint program via CPI
     ///
-    /// This function demonstrates how a transceiver program would interact
-    /// with the router program to pick up a message from the outbox.
+    /// This function demonstrates how a adapter program would interact
```
Comment (bingyuyap) [inline svm/programs/mock-adapter/src/lib.rs:18]:
```suggestion
    /// This function demonstrates how an adapter program would interact
```

File: svm/programs/mock-integrator/tests/exec_message.rs [hunk 7]
```diff
@@ -42,35 +42,33 @@ async fn setup_test_environment() -> (ProgramTestContext, Keypair, Keypair, Pubk
...
-    let (transceiver_pda, _) =
-        Pubkey::find_program_address(&[b"transceiver_pda"], &transceiver_program_id);
+    // Setup adapter
+    let adapter_program_id = mock_adapter::id();
+    let (adapter_info_pda, _) = AdapterInfo::pda(&integrator_program_id, &adapter_program_id);
+    let (adapter_pda, _) = Pubkey::find_program_address(&[b"adapter_pda"], &adapter_program_id);
 
-    // Add and enable transceiver
-    add_transceiver(
+    // Add and enable adapter
```
Comment (bingyuyap) [inline svm/programs/mock-integrator/tests/exec_message.rs:50]:
```suggestion
    // Add an enable adapter
```

Comment (evan-gray) [reply svm/programs/mock-integrator/tests/exec_message.rs:50]: existing typo, let's change separately

File: svm/programs/mock-adapter/src/lib.rs (hunk 6 above)
Comment (evan-gray) [reply svm/programs/mock-adapter/src/lib.rs]: fixed

File: svm/programs/mock-adapter/README.md (hunk 5 above)
Comment (evan-gray) [reply svm/programs/mock-adapter/README.md]: fixed

File: svm/programs/endpoint/src/lib.rs (hunk 4 above)
Comment (evan-gray) [reply svm/programs/endpoint/src/lib.rs]: fixed

File: aptos/endpoint/sources/integrator.move [hunk 8]
```diff
@@ -137,74 +137,74 @@ module router::integrator {
...
-        let len = vector::length(transceivers);
-        assert!(len < MAX_TRANSCEIVERS, E_MAX_TRANSCEIVERS_REACHED);
-        // Appends the `transceiverAddr` to the Integrator's array of Transceivers. THIS IS NOT REVERSIBLE. Once a transceiver is added for an Integrator, it cannot be removed.
-        // Note: When a Transceiver is added, it is not enabled for sending or receiving on any chain.
-        vector::push_back(transceivers, transceiver_addr);
-        table::add(&mut TransceiverToIndexStore[integrator_addr].transceiver_to_index, transceiver_addr, (len as u8));
+        let len = vector::length(adapters);
+        assert!(len < MAX_ADAPTERS, E_MAX_ADAPTERS_REACHED);
+        // Appends the `adapterAddr` to the Integrator's array of Adapters. THIS IS NOT REVERSIBLE. Once an adapter is added for an Integrator, it cannot be removed.
+        // Note: When an Adapter is added, it is not enabled for sending or receiving on any chain.
```
Comment (bingyuyap) [inline aptos/endpoint/sources/integrator.move:152]: Nit: `Adapter` on line 152 starts with uppercasing but on 151 it's lowercase

Comment (evan-gray) [reply aptos/endpoint/sources/integrator.move:152]: existing casing, not sure it's particularly important to change here

**Review by bingyuyap — APPROVED** [review]
this pr got me like

image

File: README.md (hunk 3 above)
Comment (evan-gray) [reply README.md]: good catch! I think I fixed this in the latest commit.

**Review by bingyuyap — DISMISSED** [review]

File: svm/programs/endpoint/README.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# GMP Router
+# GMP Endpoint
```
Comment (bingyuyap) [inline svm/programs/endpoint/README.md:1]:
```suggestion
# Modular Messaging Endpoint
```

Comment (evan-gray) [reply svm/programs/endpoint/README.md]: thanks! fixuped

**Review by bingyuyap — DISMISSED** [review]

**Review by bruce-riley — APPROVED** [review]

**Pull request merged**